Teleport returns a proper error message if BoltDB is locked

kontsevoy opened the pull request:
Fixes #222

kontsevoy committed:
Teleport returns a proper error message if BoltDB is locked

Fixes #222

klizhentas commented on lib/backend/boltbk/boltbk.go:
```diff
@@ -76,8 +76,11 @@ func New(path string, opts ...Option) (*BoltBackend, error) {
+	db, err := bolt.Open(path, 0600, &bolt.Options{Timeout: 5 * time.Second})
 	if err != nil {
+		if err == bolt.ErrTimeout {
+			return nil, trace.Errorf("Local storage is locked. Another instance is running? (%v)", path)
```
I think `trace.ConnectionProblem` suits better here

klizhentas replied:
also, error messages should start from lower case (as they are usually prefixed with logs messages)

kontsevoy committed:
Intermediate commit

kontsevoy committed:
Added ability to specify which console to use

...by teleport clients + servers, meaning:

1. Servers do not default to stdout when printing startup messages
2. Clients can use arbitrary input/output instead of stdin/stdout when
   doing SSH/join. This helps with integration testing.

kontsevoy committed:
Addded integration tests for:

- interactive SSH (with shell)
- joining sessions
